### PR TITLE
Replace cleartext creds to secrets in sample manifests

### LINKED ISSUE #29455: Sample manifests store minio credentials in cleartext
**What happened**:

The [starter-s3.yaml](https://github.com/kubernetes/test-infra/blob/master/config/prow/cluster/starter/starter-s3.yaml) and [starter-azure.yaml](https://github.com/kubernetes/test-infra/blob/master/config/prow/cluster/starter/starter-azure.yaml) sample manifest for deploying prow includes a deployment for minio. This deployment has the minio credentials set directly as environment variable `value` (instead of through a `secretKeyRef`).
https://github.com/kubernetes/test-infra/blob/5c1e343a49703ebd0a545e3fb9b5c28d814c1b6a/config/prow/cluster/starter/starter-s3.yaml#L1317-L1321

This goes against best practices and could be considered a security issue.
A user setting up prow and minio by following the guide and using the sample manifests would store the minio credentials in cleartext.
This would allow an attacker with existing access to the cluster where minio is deployed, and with privileges to view deployments in the relevant namespace, to read the secrets. This could potentially give the attacker admin level access to minio.

References:

- [Guide for deploying prow](https://docs.prow.k8s.io/docs/getting-started-deploy)
- [Sample manifest with minio as blob storage](https://github.com/kubernetes/test-infra/blob/master/config/prow/cluster/starter/starter-s3.yaml)
- [Sample manifest with Azure blob storage](https://github.com/kubernetes/test-infra/blob/master/config/prow/cluster/starter/starter-azure.yaml)

**What you expected to happen**:

The sample manifests should not suggest insecure ways of storing the credentials. They should be stored in secrets and set through references to them, as is done in other places in the sample manifests, e.g. here:
https://github.com/kubernetes/test-infra/blob/5c1e343a49703ebd0a545e3fb9b5c28d814c1b6a/config/prow/cluster/starter/starter-s3.yaml#L982-L987

**How to reproduce it (as minimally and precisely as possible)**:

1.  Follow the guide for deploying prow: https://docs.prow.k8s.io/docs/getting-started-deploy/#update-the-sample-manifest
1.  Pick either the `starter-s3.yaml` or `starter-azure.yaml` sample manifest.
1.  Check the environment variables in the deployment named `minio`. It has credentials stored in cleartext.

**Anything else we need to know?**:

I initially went through the security issue reporting process since that is how security issues should be reported. After review, it was agreed that the impact is low enough that it an be handled publicly.

I think this could be a good first issue that should be straight forward for a new contributor to fix.

### PR DESCRIPTION
**Pull request by sarab97**
Fixes #29455

### CODE REVIEW

File: config/prow/cluster/starter/starter-azure.yaml [hunk 1]
```diff
@@ -1286,9 +1286,15 @@ spec:
         - --console-address=:"<<CHANGE_ME_MINIO_CONSOLE_PORT>>"
         env:
         - name: MINIO_ROOT_USER
-          value: "<<CHANGE_ME_AZURE_STORAGE_ACCOUNT_USER>>"
+          valueFrom:
+            secretKeyRef:
+              name: minio-root-creds
+              key: user
```
Comment (dims) [inline config/prow/cluster/starter/starter-azure.yaml:1292]: `user` is specific to an installation (line 1292), please use `<<` and `>>` to indicate as such

File: config/prow/cluster/starter/starter-azure.yaml [hunk 2]
```diff
@@ -1286,9 +1286,15 @@ spec:
...
         - name: MINIO_ROOT_PASSWORD
-          value: "<<CHANGE_ME_AZURE_STORAGE_ACCOUNT_PASSWORD>>"
+          valueFrom:
+            secretKeyRef:
+              name: minio-root-creds
+              key: password
```
Comment (dims) [inline config/prow/cluster/starter/starter-azure.yaml:1297]: same as above

File: config/prow/cluster/starter/starter-azure.yaml (hunk 1 above)
Comment (sarab97) [reply config/prow/cluster/starter/starter-azure.yaml:1292]: this is a key to read the value from secret. This should not be edited. Thats why its not enclosed with `<<>>`

**Comment by MaxHorstmann** [comment]
@dims @sarab97 hi, still planning on merging this?

Note also https://github.com/kubernetes/test-infra/pull/29935

**Comment by sarab97** [comment]
oh yes @MaxHorstmann. Still waiting on reply from @dims. I followed the currently implemented approach. However the PR you linked does suggest better approach Im open to work to implement that here as well. 
@dims can you comment?

**Comment by MaxHorstmann** [comment]
@sarab97 thanks for your response! fyi #29935 now merged.

**Review by cblecker — COMMENTED** [review]
You're referencing a secret that doesn't exist. I'd suggest creating a new secret object in each of these templates and move the CHANGE ME reference there.

**Pull request closed without merging**